specutils/fitting/fitmodels.py

```diff
@@ -1,20 +1,246 @@
 from ..spectra.spectral_region import SpectralRegion
 from ..spectra.spectrum1d import Spectrum1D
 from astropy.modeling import fitting, Model, models
+from astropy.table import QTable
+
+
+__all__ = ['find_lines_threshold', 'find_lines_derivative', 'fit_lines', 
+           'estimate_line_parameters']
+
+# Define the initial estimators
+#   This is the default methods to use to estimate astropy model
+#   parameters. This is based on only a small subset of the astropy
+#   models but it was determined that this is a decent start as most
+#   fitting will probably use one of these.
+#
+#   Each method list must take a Spectrum1D object and should return
+#   a Quantity.
+
+_parameter_estimators = {
+    'Gaussian1D': {
+        'amplitude': lambda s: max(s.flux),
+        'mean': lambda s: centroid(s, region=None),
+        'stddev': lambda s: fwhm(s) 
+    },
+    'Lorentz1D': {
+        'amplitude': lambda s: max(s.flux),
+        'x_0': lambda s: centroid(s, region=None),
+        'fwhm': lambda s: fwhm(s) 
+    },
+    'Voigt1D': {
+        'x_0': lambda s: centroid(s, region=None),
+        'amplitude_L': lambda s: max(s.flux),
+        'fwhm_L': lambda s: fwhm(s) / np.sqrt(2),
+        'fwhm_G': lambda s: fwhm(s) / np.sqrt(2) 
+    }
+}
+
+
+def _set_parameter_estimators(model):
+    """
+    Helper method used in method below.
+    """
+    if model.__class__.__name__ in _parameter_estimators:
+        model._constraints['parameter_estimator'] = _parameter_estimators[model.__class__.__name__]
+    return model
+
+
+def estimate_line_parameters(spectrum, model):
+    """
+    The input ``model`` parameters will be estimated from the input ``spectrum``. The
+    ``model`` can be specified with default parameters, for example ``Gaussian1D()``.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.Spectrum1D`
+        The spectrum object from which we will estimate the model parameters.
+
+    model : `~astropy.modeling.Model`
+        Model for which we want to estimate parameters from the spectrum.
+
+    Returns
+    -------
+    model : `~astropy.modeling.Model`
+        Model with parameters estimated.
+    """
+
+    if not 'parameter_estimator' in model._constraints:
+        model = _set_parameter_estimators(model)
+
+    # Estimate the parameters based on the estimators already
+    # attached to the model
+    if 'parameter_estimator' in model._constraints:
+        for param, estimator in model._constraints['parameter_estimator'].items():
+            setattr(model, param, estimator(spectrum))
+
+    # No estimators.
+    else:
+        raise Exception('No method to estimate parameters')
+
+    return model
+
+
+def _consecutive(data, stepsize=1):
+    return np.split(data, np.where(np.diff(data) != stepsize)[0]+1)
+
+
+def find_lines_threshold(spectrum, noise_factor=1):
+    """
+    Find the emission and absorption lines in a spectrum. The method
+    here is a very simple threshold based on the standard deviation
```

Comment:
this is no longer based on the stdev, right? Shouldn't it be something like "base on deviations significantly larger than the spectrum's uncertainty?

Reply:
changed the description